File: vol2/ch21/LeetCode1062.cpp
class Solution {
public:
    int longestRepeatingSubstring(string s) {
        int n=s.size();
        auto dp=vector<vector<int>>(n,vector<int>(n,0));
        int ans=0;
        for(int i=0;i<n;i++) {
            for(int j=i+1;j<n;j++) {
                if(s[i]==s[j]) {
                    if(i==0) dp[i][j]=1;
                    else dp[i][j]=dp[i-1][j-1]+1;
                }
                ans=max(ans,dp[i][j]);
            }
        }
        return ans;
    }
};

File: vol2/ch21/LeetCode1062.py
class Solution:
    def longestRepeatingSubstring(self, s: str) -> int:
        n=len(s)
        dp=[[0 for j in range(n)] for i in range(n)]
        ans=0
        for i in range(n):
            for j in range(i+1,n):
                if s[i]==s[j]:
                    if i==0:dp[i][j]=1
                    else:dp[i][j]=dp[i-1][j-1]+1
                ans=max(ans,dp[i][j])
        return ans
